packages/dev/core/src/Culling/ray.ts

```diff
@@ -838,6 +845,10 @@ Scene.prototype._internalPick = function (
         }
     }
 
+    if (oldCamera) {
+        this._activeCamera = oldCamera;

```

Comment:
You will probably need to call `computeWorldMatrix(true)` for all the meshes we have looped on above, to make sure their world matrices are in line with the new active camera (just in case they are used in the reset of the frame). However, if you fix it at the `getWorldMatrix / computeWorldMatrix` level (as described above) you won't need to do that :)